### DISCUSSION TITLE
Compiling on Windows - Visual Studio 2022

### DISCUSSION THREAD

**Discussion by sramey** [FundamentalFrequency/.github#35] [General]
FYI: 

I was able to compile on Windows with Visual Studio 2022, C++ tools installed and CMAKE component

It was was easier than I thought. You right click the folder and select open in Visual Studio

I had to change one thing in regards to YAML in LMN3-DAW CmakeSettings and Open Cmakelists.txt and comment out atomic on line 194

![image](https://user-images.githubusercontent.com/8069/176731679-020b4a19-a44c-4eb1-9786-7e4b88df3de2.png)

After building I also had to copy over yaml-cppd.dll from the YAML-CPP build, into where LMN3.exe was located (I'm sure you can configure something in VS to copy it over)

![image](https://user-images.githubusercontent.com/8069/176732159-423d5eca-c71a-4852-bcb2-cfcd59f01710.png)

The emulator makes sounds when the buttons are pressed but it doesn't send it to the DAW for some windows reason. I'll try to figure that out.

**Comment by stonepreston** [comment]
On linux, the default software midi input/output can essentially be used as a loopback device, meaning you can send and receive on the same software midi device. This is how the emulator communicates with the DAW, it sends data out on that device, while the DAW receives on it. 

You are going to need to find some way to do this in Windows. Ive seen some folks using Mac achieve this using "Virtual midi devices" so maybe theres something similar to that in Windows

**Comment by sramey** [comment]
I was able to get everything working (Emulator connecting to the DAW) in Windows.

I had to install loopMIDI (Free) to setup a windows loopback midi port
https://www.tobias-erichsen.de/software/loopmidi.html

Everything built and works perfect from VS. Can't wait to dig in.

**Comment by coderoid** [comment]
@sramey regarding the change you made to yaml what was the change specifically, I am getting this work with vs code instead of vs 2022. And the change wasn't obvious to me looking at the screenshot you posted.
